apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.DatasetCrud.cs

```diff
@@ -179,12 +187,20 @@ private ListRequest CreateListDatasetsRequest(ProjectReference projectReference,
             return request;
         }
 
-        private InsertRequest CreateInsertDatasetRequest(DatasetReference datasetReference, CreateDatasetOptions options)
+        private InsertRequest CreateInsertDatasetRequest(DatasetReference datasetReference, Dataset resource, CreateDatasetOptions options)
         {
             GaxPreconditions.CheckNotNull(datasetReference, nameof(datasetReference));
-            var dataset = new Dataset { DatasetReference = datasetReference, Location = DefaultLocation };
-            var request = Service.Datasets.Insert(dataset, datasetReference.ProjectId);
-            options?.ModifyRequest(dataset, request);
+            GaxPreconditions.CheckArgument(
+                resource?.DatasetReference == null || datasetReference.Equals(resource.DatasetReference),
+                nameof(resource.DatasetReference),
+                $"If {nameof(resource.DatasetReference)} is specified, it must be the same as {nameof(datasetReference)}");
+
+            resource ??= new Dataset { DatasetReference = datasetReference, Location = DefaultLocation };
```

Comment:
Just use new `Dataset()` here and let the following lines handle the defaults? Avoids code repetition.